### PR TITLE
[TSK-227] add kafka support for transaction manager report generation

### PR DESCRIPTION
**Pull request by kostya4201337**

### CODE REVIEW

File: notification-manager/src/main/java/com/qpay/notificationmanager/generator/impl/MailMessageGenerator.java [hunk 1]
```diff
@@ -25,4 +33,20 @@ public SimpleMailMessage getPaymentMessage(final PaymentNotification paymentNoti
...
+        return mimeMessage -> {
+            final var templateMessage = emailTemplate.getReport();
+            final var text = format(templateMessage.getText(), reportNotification.dateFrom(), reportNotification.dateTo());
+            mimeMessage.setRecipient(Message.RecipientType.TO, new InternetAddress(reportNotification.email()));
+            mimeMessage.setSubject(templateMessage.getSubject());
+            mimeMessage.setText(text);
+
+            final var file = new File(reportNotification.filePath() + reportNotification.fileName());
+            final FileSystemResource fileSystemResource = new FileSystemResource(file);
+            final MimeMessageHelper mimeMessageHelper = new MimeMessageHelper(mimeMessage, true);
```
Comment (adsukharev) [inline notification-manager/src/main/java/com/qpay/notificationmanager/generator/impl/MailMessageGenerator.java]: var

File: notification-manager/src/main/java/com/qpay/notificationmanager/service/impl/MockNotificationService.java [hunk 2]
```diff
@@ -17,4 +18,9 @@ public void sendMessage(final PaymentNotification paymentNotification) {
         log.info("Mock email message sent to email: {}", paymentNotification.emailTo());
     }
 
+    @Override
+    public void sendMessage(final ReportNotification reportNotification) {
+        log.info("Mock email message sent to email: {}", reportNotification.email());
```
Comment (adsukharev) [inline notification-manager/src/main/java/com/qpay/notificationmanager/service/impl/MockNotificationService.java]:
```suggestion
        log.info("Mock report sent to email: {}", reportNotification.email());
```

File: notification-manager/src/main/resources/application-local.yaml [hunk 3]
```diff
@@ -18,5 +18,7 @@ email:
 broker:
   groups:
     payment-group: local_payment_group_1
+    transaction-group: local_transaction_group1
```
Comment (adsukharev) [inline notification-manager/src/main/resources/application-local.yaml]:
```suggestion
    report-group: local_report_group1
```

File: notification-manager/src/main/resources/application.yaml [hunk 4]
```diff
@@ -28,12 +28,17 @@ email-template:
   payment:
     subject: "Payment Notification"
     text: "You received a payment of %s. Thank you for using our service."
+  report:
+    subject: "Transaction Report"
```
Comment (adsukharev) [inline notification-manager/src/main/resources/application.yaml]:
```suggestion
    subject: "Transactions Report"
```

File: transaction-history-manager/src/main/java/com/qpay/transactionhistorymanager/event/ReportNotificationProducer.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+public class ReportNotificationProducer {
+
+    private final KafkaTemplate<String, ReportNotification> kafkaTemplate;
+
+    @Value("${broker.topics.report-topic}")
+    private String paymentTopic;
+
+    public void publishReportNotification(final ReportNotification reportNotification) {
+        kafkaTemplate.send(paymentTopic, reportNotification.email(), reportNotification);
+        log.info("Payment notification event sent: {}", reportNotification);
```
Comment (adsukharev) [inline transaction-history-manager/src/main/java/com/qpay/transactionhistorymanager/event/ReportNotificationProducer.java]:
```suggestion
        log.info("Report notification event sent: {}", reportNotification);
```

File: transaction-history-manager/src/main/java/com/qpay/transactionhistorymanager/event/ReportNotificationProducer.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+@Slf4j
+public class ReportNotificationProducer {
+
+    private final KafkaTemplate<String, ReportNotification> kafkaTemplate;
+
+    @Value("${broker.topics.report-topic}")
+    private String paymentTopic;
+
+    public void publishReportNotification(final ReportNotification reportNotification) {
+        kafkaTemplate.send(paymentTopic, reportNotification.email(), reportNotification);
```
Comment (adsukharev) [inline transaction-history-manager/src/main/java/com/qpay/transactionhistorymanager/event/ReportNotificationProducer.java]: if you copy code, change names

File: notification-manager/src/main/java/com/qpay/notificationmanager/service/impl/EmailNotificationService.java [hunk 7]
```diff
@@ -25,4 +26,8 @@ public void sendMessage(final PaymentNotification paymentNotification) {
         log.info("Message sent to: {}", paymentNotification.emailTo());
     }
 
+    public void sendMessage(final ReportNotification reportNotification) {
+        final var message = mailMessageGenerator.getReportMessage(reportNotification);
+        mailSender.send(message);
```
Comment (adsukharev) [inline notification-manager/src/main/java/com/qpay/notificationmanager/service/impl/EmailNotificationService.java:31]: log.info("Report sent to: {}", paymentNotification.emailTo());

**Review by adsukharev — APPROVED** [review]
I checked the code briefly. Let Roma look at it in detail.

**Comment by adsukharev** [comment]
while Roma is reviewing your PR, you can update UI (if you have any ideas) and work on frontend features (top up and profile)

**Review by Hoocki — APPROVED** [review]

**Pull request merged**